Content.Server/Atmos/EntitySystems/AtmosphereSystem.Monstermos.cs

```diff
@@ -584,20 +584,31 @@ private void ConsiderFirelocks(
             foreach (var entity in _map.GetAnchoredEntities(ent.Owner, mapGrid, tile.GridIndices))
             {
                 if (_firelockQuery.TryGetComponent(entity, out var firelock))
-                    reconsiderAdjacent |= _firelockSystem.EmergencyPressureStop(entity, firelock);
+                    reconsiderAdjacent |= _firelockSystem.EmergencyPressureStopAirtight((entity, firelock));
             }
 
             foreach (var entity in _map.GetAnchoredEntities(ent.Owner, mapGrid, other.GridIndices))
             {
                 if (_firelockQuery.TryGetComponent(entity, out var firelock))
-                    reconsiderAdjacent |= _firelockSystem.EmergencyPressureStop(entity, firelock);
+                    reconsiderAdjacent |= _firelockSystem.EmergencyPressureStopAirtight((entity, firelock));
             }
 
             if (!reconsiderAdjacent)
                 return;
 
+            // Before updating the adjacent tile flags that determine whether air is allowed to flow
+            // or not, we explicitly update airtight data on these tiles right now.
+            // This ensures that UpdateAdjacentTiles has updated data before updating flags.
+            // This allows monstermos' floodfill check that determines if firelocks have dropped
+            // to work correctly.
+            var tilegridAtmosTile = GetOrNewTile(ent.Owner, ent.Comp1, tile.GridIndices);
+            var othergridAtmosTile = GetOrNewTile(ent.Owner, ent.Comp1, other.GridIndices);
```

Comment:
What is the difference between getting a new tile here and just using `tile` and `other` directly? Cursory testing seems to show it results in the same behavior.